plugin/src/main/kotlin/com/exactpro/th2/gradle/PublishTh2Plugin.kt

```diff
@@ -55,6 +55,11 @@ class PublishTh2Plugin : Plugin<Project> {
         check(project === project.rootProject) {
             "th2 publish plugin must be applied to the root project but was applied to ${project.path}"
         }
+        project.gradle.projectsEvaluated {
+            check(project.allprojects.any { it.plugins.hasPlugin(MavenPublishPlugin::class.java) }) {
+                "Plugin 'maven-publish' is not applied to any of ${project.allprojects.map(Project::getName)} projects"
```

Comment:
issue: This error does not explain why this is a problem. If I saw this error, I would have no idea how to fix it. Let's add explanation
```suggestion
                "'th2.gradle.publish' plugin is applied to the root project, but none of the projects has 'maven-publish' plugin to enable the publication. You can either remove 'th2.gradle.publish' plugin from the root project or add the 'maven-publish' plugin to projects you want to publish as artifacts"
```